src/arithmetic/agg_funcs.c

```diff
@@ -93,6 +126,31 @@ int __agg_avgStep(AggCtx *ctx, SIValue *argv, int argc) {
 	return AGG_OK;
 }
 
+int __agg_avgDistinctStep(AggCtx *ctx, SIValue *argv, int argc) {
+	// convert the value of the input sequence to a double if possible
+	__agg_avgCtx *ac = Agg_FuncCtx(ctx);
+
+	double n;
+	for(int i = 0; i < argc; i ++) {
```

Comment:
I know each `Step` function accepts a list of arguments `argv` but we've always passed just a single value, `argc = 1`.
if we change the signature from `Step(AggCtx *ctx, SIValue *argv, int argc)` to `Step(AggCtx *ctx, SIValue v)`
the `distinct` version becomes even shorter:
```
StepDistincr(AggCtx *ctx, SIValue v) {
    // Check if `v` is in set, if not call Step function
    // Check Step function return value, if no error occurred, add `v` to set.
}
```
In addition this will save looping over a single value...
Thoughts?

Reply:
:+1: to this!

On the subject of outmoded aggregation logic, the complaints registered with `Agg_SetError` are never emitted and do not modify the output:
```
$ redis-cli GRAPH.QUERY G  "UNWIND [1,'f',5] AS a RETURN SUM(DISTINCT a)"
1) 1) "SUM(DISTINCT a)"
2) 1) 1) "6"
```
We should update this area to use the same run-time error logic we've introduced for non-aggregate functions. (Not necessarily in this PR, of course)

Reply:
👍 agree